### ISSUE TITLE
Missing fields annotation false positive

### ISSUE THREAD

**Issue by alygin** [intellij-rust/intellij-rust#941]
`struct`s might have configuration-dependent fields which results in false positive "Some fields are missing" errors because we ignore `cfg` attributes. Here's an example (from `ripgrep`):

```rust
pub struct BufferWriter {
    stdout: io::Stdout,
    printed: AtomicBool,
    separator: Option<Vec<u8>>,
    color_choice: ColorChoice,
    #[cfg(windows)]
    console: Option<Mutex<wincolor::Console>>,
}
```

which leads to an incorrect error in the following code:

```rust
#[cfg(not(windows))]
pub fn stdout(choice: ColorChoice) -> BufferWriter {
    BufferWriter {
        stdout: io::stdout(),
        printed: AtomicBool::new(false),
        separator: None,
        color_choice: choice,
    }
 // ^ Error: Some fields are missing
}
```

I think the error annotator should skip fields with the `cfg`-attribute (like it's done in the name duplication analysis). Even broader, the `cfg` attribute must be checked all the way up to the crate root because it's the whole module that can be configuration-dependent. Or we can find some reasonable level to stop at, like the enclosing module.

Or maybe it's an exaggeration and such deep searches are not necessary because the `struct` won't be resolved in the following case?

```rust
#[cfg(windows)] struct Chameleon { win_field: u32 }
#[cfg(not(windows))] struct Chameleon { non_win_field: u32 }
```

Some other error annotations might also be prone to the same false-positiveness.
